src/peft/utils/hotswap.py

```diff
@@ -13,23 +13,302 @@
+
+    for module in model.modules():
+        # Decide which pad function to call based on module type
+        if isinstance(module, Linear):
+            pad_fn = _get_padded_linear
+        elif isinstance(module, Conv2d):
+            pad_fn = _get_padded_conv2d
+        else:
+            # Skip any other module types
+            continue
+
+        # Pad LoRA A
+        for adapter_name, lora_A_module in module.lora_A.items():
+            new_layer = pad_fn(lora_A_module, target_rank=target_rank, is_lora_A=True)
+            module.lora_A[adapter_name] = new_layer
+
+        # Pad LoRA B
+        for adapter_name, lora_B_module in module.lora_B.items():
+            new_layer = pad_fn(lora_B_module, target_rank=target_rank, is_lora_A=False)
+            module.lora_B[adapter_name] = new_layer
+
+
+def prepare_model_for_compiled_hotswap(
+    model: torch.nn.Module,
+    *,
+    target_rank: Optional[int] = None,
+    config: Optional[LoraConfig | dict[str, LoraConfig]] = None,
+) -> None:
+    """
+    Helper function that prepares the model so that it can later be compiled and then used with hot-swapping.
+
+    It is necessary to call this function on the model for hot-swapping to work if both of these are true:
+
+    - the different LoRA adapters have different ranks and/or different alpha values (i.e. scalings)
+    - you plan to torch.compile the model and want to avoid re-compilation
+
+    It is important to call this function *after* the first LoRA adapter has been loaded (i.e. the one that will be
+    swapped out) but *before* the model is compiled.
+
+    Even with this function, hot-swapping LoRA adapters that target different layers is still not supported.
+
+    Note: This function modifies the model in-place. If you want to restore the model to its initial state, you will
+    have to reload it.
+
+    Args:
+        model (`nn.Module`):
+            The model with the loaded adapter, before compilation.
+        target_rank (`int`, *optional*):
+            The target rank to pad the LoRA weights to. Should be the maximum rank among all LoRA adapters that will be
+            hot-swapped. If not specified, the target ranks will not be changed.
+        config (`LoraConfig` or `dict[str, LoraConfig]`, *optional*):
+            Optionally pass the `LoraConfig`s of the LoRA adapters. If passed, the rank in the configs will be updated
+            to `target_rank`.
+    """
+    is_compiled = hasattr(model, "_orig_mod")
+    if is_compiled:
+        raise ValueError("Call prepare_model_for_compiled_hotswap *before* compiling the model")
+
+    _convert_scalings_to_tensor(model)
+    if target_rank is not None:
+        _pad_lora_weights(model, target_rank=target_rank)
+
+    if not config:
+        return
+    if target_rank is None:
+        return
+
+    if not isinstance(config, dict):
+        # config can be either a PeftConfig, or a dict of PeftConfigs like PeftModel.peft_config
+        config = {"dummy": config}
```

Comment:
This config won't have any repercussions because it's never assigned to a `PeftModel`, right?